go/bundle/version/version.go

```diff
@@ -44,31 +56,39 @@ func (v Version) HeaderMagicBytes() []byte {
 	case Unversioned:
 		return HeaderMagicBytesUnversioned
 	case VersionB1:
-		return append(HeaderMagicBytes, VersionMagicBytesB1...)
+		return append(HeaderMagicBytesB1, VersionMagicBytesB1...)
+	case VersionB2:
+		return append(HeaderMagicBytesB2, VersionMagicBytesB2...)
 	default:
 		panic("not reached")
 	}
 }
 
 func ParseMagicBytes(r io.Reader) (Version, error) {
-	hdrMagic := make([]byte, len(HeaderMagicBytes))
+	hdrMagic := make([]byte, len(HeaderMagicBytesB1))
 	if _, err := io.ReadFull(r, hdrMagic); err != nil {
 		return "", err
 	}
 	if bytes.Compare(hdrMagic, HeaderMagicBytesUnversioned) == 0 {
 		return Unversioned, nil
 	}
-	if bytes.Compare(hdrMagic, HeaderMagicBytes) != 0 {
-		return "", errors.New("bundle: header magic mismatch")
-	}
 
 	verMagic := make([]byte, len(VersionMagicBytesB1))
 	if _, err := io.ReadFull(r, verMagic); err != nil {
 		return "", err
 	}
 	if bytes.Compare(verMagic, VersionMagicBytesB1) == 0 {
+		if bytes.Compare(hdrMagic, HeaderMagicBytesB1) != 0 {
```

Comment:
Can we test the header bytes first, and then the version bytes?

`dump-bundle -i <non-wbn-file>` should fail with "header magic mismatch" rather than "unrecognized version magic".